R/read_sourmash_outputs.R

```diff
@@ -1,3 +1,20 @@
+#' Check if local file path or URL are valid
+#'
+#' @param file File path or URL
+#'
+#' @return NULL
+#'
+#' @examples
+check_file_and_url_one <- function(file){
+  # check if file or URL are valid, and if not, print helpful messages and errors.
+  if(!file.exists(file)){
```

Comment:
It's generally not great to make these checks in a nested way, because it can get tough to follow for the reader (let's say you add a check for S3 URIs in the future, it'd have to be nested one-level deeper). 

The heuristics I use for this type of stuff is: (a) it's better to return early from validation methods and (b) following that, checking for something is valid (rather than not valid) leads to more linear checks, rather than more nested checks

With these heuristics, here's how I would write this method (it's more verbose, but clearer imo):

```
check_file_and_url_one <- function(file){
  # check if file or URL are valid, and if not, print helpful messages and errors.
  if (file.exists(file)) {
    return(NULL)
  }

  message("File does not exist locally. Checking as url. If you did not pass a url, please provide a valid file path and retry. The Sys.glob() function might help if you're specifying multiple file paths.")
  if(httr::HEAD(file)$status == 200) {
    return(NULL)
  }

  stop("Neither file path nor url are valid.")
}
```

Reply:
ooooh i love this. I didn't think of doing a `return(NULL)`, and that solves a lot of problems I was having with the logic of this. I'll make this change before merging!